core/kernel/ldelf_syscalls.c

```diff
@@ -133,17 +134,21 @@ TEE_Result ldelf_syscall_open_bin(const TEE_UUID *uuid, size_t uuid_size,
 		sess->user_ctx = sys_ctx;
 	}
 
+	res = memdup_user(uuid, uuid_size, (void **)&uuid_kbuf);
```

Comment:
Remove `vm_check_access_rights()` at line 113 as `memdup_user()` already takes care of this sanity test.